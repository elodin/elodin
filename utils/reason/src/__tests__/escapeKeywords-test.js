import { parse } from '@elodin/core'

import escapeKeywords from '../escapeKeywords'

describe('Escaping keywords', () => {
  it('should return a capitalized string', () => {
    const { ast } = parse(`
variant Type {
  Primary
  Secondary
}

style Button {
  color: $color
  backgroundColor: $bgColor
  
  [Type=Primary] {
    color: blue
  }
}
`)

    expect(
      escapeKeywords(ast, ['Type', 'Secondary', 'color'])
    ).toMatchSnapshot()
  })
})
