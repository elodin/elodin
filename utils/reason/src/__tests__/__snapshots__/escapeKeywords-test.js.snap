// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Escaping keywords should return a capitalized string 1`] = `
Object {
  "body": Array [
    Object {
      "body": Array [
        Object {
          "comments": Array [],
          "type": "Identifier",
          "value": "Primary",
        },
        Object {
          "comments": Array [],
          "type": "Identifier",
          "value": "Secondary_",
        },
      ],
      "comments": Array [],
      "name": "Type_",
      "type": "Variant",
    },
    Object {
      "body": Array [
        Object {
          "comments": Array [],
          "dynamic": true,
          "property": "color",
          "raw": false,
          "type": "Declaration",
          "value": Object {
            "environment": false,
            "type": "Variable",
            "value": "color_",
          },
        },
        Object {
          "comments": Array [],
          "dynamic": true,
          "property": "backgroundColor",
          "raw": false,
          "type": "Declaration",
          "value": Object {
            "environment": false,
            "type": "Variable",
            "value": "bgColor",
          },
        },
        Object {
          "body": Array [
            Object {
              "comments": Array [],
              "dynamic": false,
              "property": "color",
              "raw": false,
              "type": "Declaration",
              "value": Object {
                "type": "Identifier",
                "value": "blue",
              },
            },
          ],
          "comments": Array [],
          "operator": "=",
          "property": Object {
            "type": "Identifier",
            "value": "Type_",
          },
          "type": "Conditional",
          "value": Object {
            "type": "Identifier",
            "value": "Primary",
          },
        },
      ],
      "comments": Array [],
      "name": "Button",
      "type": "Style",
    },
  ],
  "type": "File",
}
`;
